src/main/java/org/primefaces/component/panelgrid/PanelGridRenderer.java

```diff
@@ -305,8 +305,10 @@ public void encodeGridBody(FacesContext context, PanelGrid grid, int columns, St
             }
         }
 
-        if (i != 0 && (i % columns) != 0) {
-            writer.endElement("div");
+        if (LAYOUT_GRID.equalsIgnoreCase(layout)) {
+            if (i != 0 && (i % columns) != 0) {
+                writer.endElement("div");
+            }
```

Comment:
I think you can remove this fix as Mert just refactored the panelGrid for Flex option in https://github.com/primefaces/primefaces/issues/5275